router/network.go

```diff
@@ -19,108 +19,18 @@ type NetHandleT struct {
+func (network *NetHandleT) sendPost(jsonData []byte) (statusCode int, status string, respBody string) {
 	client := network.httpClient
 	//Parse the response to get parameters
-	postInfo := integrations.GetPostInfo(jsonData)
-
-	requestConfig, ok := postInfo.RequestConfig.(map[string]interface{})
-	if !ok {
-		panic(fmt.Errorf("typecast of postInfo.RequestConfig to map[string]interface{} failed"))
-	}
-	requestMethod, ok := requestConfig["requestMethod"].(string)
-	if !(ok && (requestMethod == "POST" || requestMethod == "GET")) {
-		panic(fmt.Errorf("typecast of requestConfig[\"requestMethod\"] to string failed. or requestMethod:%s is neither POST nor GET", requestMethod))
-	}
-	requestFormat := requestConfig["requestFormat"].(string)
-	if !ok {
-		panic(fmt.Errorf("typecast of requestConfig[\"requestFormat\"] to string failed"))
-	}
-
-	switch requestFormat {
-	case "PARAMS":
-		postInfo.Type = integrations.PostDataKV
-	case "JSON":
-		postInfo.Type = integrations.PostDataJSON
-	default:
-		panic(fmt.Errorf("requestFormat:%s is neither PARAMS nor JSON", requestFormat))
-	}
-
-	var req *http.Request
-	var err error
-	if useTestSink {
-		req, err = http.NewRequest(requestMethod, testSinkURL, nil)
-		if err != nil {
-			panic(err)
-		}
-	} else {
-		req, err = http.NewRequest(requestMethod, postInfo.URL, nil)
-		if err != nil {
-			panic(err)
-		}
-	}
-
-	queryParams := req.URL.Query()
-	if postInfo.Type == integrations.PostDataKV {
-		payloadKV, ok := postInfo.Payload.(map[string]interface{})
-		if !ok {
-			panic(fmt.Errorf("typecast of postInfo.Payload to map[string]interface{} failed"))
-		}
-		for key, val := range payloadKV {
-			queryParams.Add(key, fmt.Sprint(val))
-		}
-	} else if postInfo.Type == integrations.PostDataJSON {
-		payloadJSON, ok := postInfo.Payload.(map[string]interface{})
-		if !ok {
-			panic(fmt.Errorf("typecast of postInfo.Payload to map[string]interface{} failed"))
-		}
-		jsonValue, err := json.Marshal(payloadJSON)
-		if err != nil {
-			panic(err)
-		}
-		req.Body = ioutil.NopCloser(bytes.NewReader(jsonValue))
-	} else {
-		//Not implemented yet
-		panic(fmt.Errorf("postInfo.Type : %d is not implemented", postInfo.Type))
-	}
-
-	req.URL.RawQuery = queryParams.Encode()
-
-	headerKV, ok := postInfo.Header.(map[string]interface{})
-	if !ok {
-		panic(fmt.Errorf("typecast of postInfo.Header to map[string]interface{} failed"))
-	}
-	for key, val := range headerKV {
-		req.Header.Add(key, val.(string))
-	}
-
-	req.Header.Add("User-Agent", "RudderLabs")
-
-	resp, err := client.Do(req)
-
-	var respBody []byte
-
-	if resp != nil && resp.Body != nil {
-		respBody, _ = ioutil.ReadAll(resp.Body)
-		defer resp.Body.Close()
-	}
-
+	postInfo, err := integrations.GetPostInfo(jsonData)
 	if err != nil {
-		logger.Error("Errored when sending request to the server", err)
-		return http.StatusGatewayTimeout, "", string(respBody)
+		panic(err)
```

Comment:
Should we return an error with custom message, instead of a panic ? We can use backed up aborted rt_jobs and replay them if needed.

Although each individual transformer builds response upon this https://github.com/rudderlabs/rudder-transformer/blob/master/v0/destinations/util.js#L79-L94, it might be flaky to build on guarantee that the response from transfomer will always have queryParams, files etc.

Reply:
Maybe we can add  the check to verify all the postInfo are in the response to transformer .
Check in versioedRouter.js before sending response and throw an error in transformer if not

Reply:
@psrikanth88 Till we enable backing up aborted jobs by default and the restream functionality might be safe to panic. Another option is we can enable backing up aborted jobs by default and then return an error instead of panic.

Reply:
@psrikanth88 @sivashanmukh WDYT?

Reply:
```
Another option is we can enable backing up aborted jobs by default and then return an error instead of panic.
```

Reply:
Return an error in VersionedRouter if the format doesn't match.